x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/cluster/routing/allocation/mapper/DataTierFieldMapper.java

```diff
@@ -53,7 +53,7 @@ protected boolean matches(String pattern, boolean caseInsensitive, SearchExecuti
                 pattern = Strings.toLowercaseAscii(pattern);
             }
             String tierPreference = DataTierAllocationDecider.INDEX_ROUTING_PREFER_SETTING.get(context.getIndexSettings().getSettings());
-            if (tierPreference == null) {
+            if (Strings.hasText(tierPreference) == false) {
```

Comment:
This check is a little unusual, but it matches how we test for the setting in other places like `DataTierAllocationDecider`.